### DISCUSSION TITLE
How do I make the babel plugin apply to node_modules?

### DISCUSSION THREAD

**Discussion by nandorojo** [software-mansion/react-native-reanimated#1715] [Q&A] [answered]
Hi there, I'm the author of [Moti](https://github.com/nandorojo/moti), a framer-motion-like library based on Reanimated 2.

We're having an issue [here](https://github.com/nandorojo/moti/issues/14) where Moti is not working on Expo web.

![image](https://user-images.githubusercontent.com/13172299/107804287-b60c0c00-6d31-11eb-90dd-cf4a630d782c.png)

## The problem

The problem is, Moti uses `useAnimatedStyle` from Reanimated under the hood. However, Reanimated's babel plugin is **not** applying to Moti, presumably since `node_modules` are ignored by Babel.

Is there a way I can include `node_modules/@motify/core` and `node_modules/moti` in Reanimated's babel plugin so that the plugin works properly?

## Workaround

Currently, I've fixed it by using `babel-loader` in `webpack.config.js`:

```js
const createExpoWebpackConfigAsync = require("@expo/webpack-config");

module.exports = async function (env, argv) {
  const config = await createExpoWebpackConfigAsync(env, argv);

  config.module.rules.push({
    test: /\.(js|ts|tsx)$/,
    exclude: /node_modules\/(?!(moti|@motify)\/).*/,
    use: "babel-loader",
  });

  return config;
};
```

I have a feeling this isn't the right solution, but I don't really know my way around Webpack/Babel.

Any help here would be greatly appreciated, thanks!

## Added context

I'm using Bob to compile Moti, and it looks like the `lib/module` folder is what Expo uses on Web.

**Comment by nandorojo** [comment, marked as answer]
I found the right answer, thanks to the `@expo/webpack-config` docs:

```js
const createExpoWebpackConfigAsync = require("@expo/webpack-config");

module.exports = async function (env, argv) {
  const config = await createExpoWebpackConfigAsync(
    {
      ...env,
      babel: { dangerouslyAddModulePathsToTranspile: ["moti"] },
    },
    argv
  );

  return config;
};
```

Expo lets you pass a custom set of node modules to transpile with `dangerouslyAddModulePathsToTranspile`, so that did it!

**Reply by IlyaTaidi** [reply]
This solution might now work anymore. Default Expo/Moti setup throwing module not found errors in web view linked repo
https://github.com/IlyaTaidi/motiPl2
